domain/consensus/model/interface_processes_transactionalidator.go

```diff
@@ -0,0 +1,10 @@
+package model
+
+// TransactionValidator exposes a set of validation classes, after which
+// it's possible to determine whether a transaction is valid
+type TransactionValidator interface {
+	ValidateTransactionInIsolation(transaction *DomainTransaction) error
+	ValidateTransactionInContext(transaction *DomainTransaction) error
+	ValidateTransactionAndCalculateFee(transaction *DomainTransaction, utxoEntries []*UTXOEntry) (fee uint64, err error)
+	ValidateTransactionAgainstUTXO(transaction *DomainTransaction) error
```

Comment:
This is not the way I envisioned these functions. See submitTransaction graph in EA, I think it explains what I ment, and if not, lets talk about this.